### PR TITLE
docs(Storybook): Leverage for component documentation

### LINKED ISSUE #2094: Update Storybook to use docs addon and dynamically generate prop tables and live examples
https://www.npmjs.com/package/@storybook/addon-docs

https://github.com/storybookjs/storybook/blob/next/addons/docs/docs/props-tables.md

- Auto-generate interface documentation and live examples with 'controls'.

- Ensure that Storybook is also supplemented with README text for components where necessary.

- Also ensure that exposed composable sub-components have their interfaces documented dynamically.

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related issue

Closes #2094

## Overview

Leverage Storybook for component technical documentation.

## Reason

>Storybook should act as a single source of truth for component technical documentation.

## Work carried out

- [x] Setup `@storybook/addon-docs`
- [x] Rework existing Stories to leverage new format
- [x] Add DocBlock comments to all component interfaces
- [x] Make actions add-on play nice with docs generation
- [x] Add badge to monorepo README
- [x] Supplement components with additional README text where appropriate
- [x] Fix some missing a11y test skips from legacy

## Screenshot

Screenshot 2021-04-22 at 13 41 35

## Developer notes

I've disabled dynamic source generation in favour of displaying the static story source. 

This Storybook feature is in early stages and needs to progress more before we can confidently adopt it.

### CODE REVIEW

File: packages/react-component-library/package.json [hunk 1]
```diff
@@ -70,11 +70,10 @@
     "@royalnavy/eslint-config-react": "^2.43.5",
     "@storybook/addon-a11y": "^6.1.9",
```
Comment (thyhjwb6) [inline packages/react-component-library/package.json]: I think it's good to keep all of the `@storybook` dependencies in sync.

**Review by thyhjwb6 — COMMENTED** [review]
Looks great.. a couple of comments:
- the a11y tab is missing
- my local instance is building every second.. do you have this?

File: packages/react-component-library/package.json (hunk 1 above)
Comment (m7kvqbe1) [reply packages/react-component-library/package.json]: The a11y tab should still be there but you have to click on 'Canvas' to see it.

**Review by thyhjwb6 — APPROVED** [review]
👍

**Review by thyhjwb6 — APPROVED** [review]
👍 

I have not read all of the documentation 😂

**Pull request merged**